analyses/chromosomal-instability/util/chr-break-plot.R

```diff
@@ -0,0 +1,118 @@
+# Functions for chromosomal instability plots
+#
+# C. Savonen for ALSF - CCDL
+#
+# 2020
+
+map_density_plot <- function(granges,
+                             y_val,
+                             y_lab,
+                             color,
+                             main_title) {
+  # Given a GRanges object, plot it y value along its chromosomal mappings using
+  # ggbio.
+  #
+  # Args:
+  #   granges: A Granges object to plot
+  #   y_val: a character string of the columnname in listData spot of the
+  #          GenomicRanges to plot on the y axis
+  #   color: a color parameter
+  #   y_lab: a character string to use for the ylabel. Will be passed to
+  #          ggplot2::ylab argument.
+  #   main_title: a character string to use for the main title.
+  #
+  # Returns:
+  #  ggplot of chromosomal mapping of the y value given.
+  #
+  # For setting the scale later, need to get y's max
+  max_y <- max(
+    data.frame(granges@elementMetadata@listData) %>%
+      dplyr::pull(
+      !!rlang::sym(y_val)
+    ),
+    na.rm = TRUE
+  )
+  # Make the density plot
+  density_plot <- ggbio::autoplot(granges, ggplot2::aes(y = !!rlang::sym(y_val)),
+    geom = "line", scales = "free_x", space = "free_x",
+    colour = color
+  ) +
+    ggplot2::theme_classic() +
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size = 3, angle = 45, hjust = 1)) +
+    ggplot2::ylab(y_lab) +
+    ggplot2::ggtitle(main_title) +
+    ggplot2::scale_y_continuous(breaks = seq(0, max_y, by = 2))
```

Comment:
Are we confident `by=2` will be universally reasonable?

Reply:
So far it is. But this is a good question to keep an eye out for.

Reply:
maybe just add a variable defining this as 2 at the top of the function, but i don't think it merits an argument.